rust_snuba/src/factory.rs

```diff
@@ -0,0 +1,139 @@
+use crate::config;
+use crate::processors;
+use crate::strategies::clickhouse::ClickhouseWriterStep;
+use crate::strategies::python::PythonTransformStep;
+use crate::types::{BadMessage, BytesInsertBatch, KafkaMessageMetadata};
+use rust_arroyo::backends::kafka::types::KafkaPayload;
+use rust_arroyo::processing::strategies::commit_offsets::CommitOffsets;
+use rust_arroyo::processing::strategies::reduce::Reduce;
+use rust_arroyo::processing::strategies::run_task_in_threads::{
+    RunTaskFunc, RunTaskInThreads, TaskRunner,
+};
+use rust_arroyo::processing::strategies::InvalidMessage;
+use rust_arroyo::processing::strategies::{ProcessingStrategy, ProcessingStrategyFactory};
+use rust_arroyo::types::{BrokerMessage, InnerMessage, Message};
+use std::sync::Arc;
+use std::time::Duration;
+
+pub struct ConsumerStrategyFactory {
```

Comment:
I noticed there are no docstrings anywhere here.
Is it intentional ?

Reply:
Mostly they are 1:1 with Snuba and Arroyo and I don't want to maintain both yet. Still haven't figured out how we'll generate shared docs for Python and Rust in the arroyo repo (the goal is both implementations will live together). But will make sure these get populated before this is actually used for real.